libc/crt/crt.S

```diff
@@ -31,6 +31,18 @@
 //	@noreturn
 _start:
 
+#if SupportsXnu()
+//	Hack for detecting M1 Rosetta environment.
+//	https://github.com/jart/cosmopolitan/issues/429#issuecomment-1166704377
+	cmp	$-1,%ebx
+	jne	0f
+	cmp	$+1,%edx
+	jne	0f
+	mov	$XNU,%cl
+	mov	$0,%rdi
```

Comment:
Is there any reason you can't avoid the need for clearing RDI by moving this piece of code below the FreeBSD check?

Reply:
Thanks for explaining! What you did sounds good then.